### PR TITLE
Adding files for a future Debian package

### PR DESCRIPTION
**Pull request by RyDroid**

### CODE REVIEW

**Comment by Bertram25** [comment]
Please have a look at this:
https://github.com/ValyriaTear/ValyriaTear/issues/216

@anholt already made a *lot* of work concerning Debian packaging. Maybe you could see any issues he had at the time?
I'd rather not accept a PR with unfinished packaging in the meanwhile, though. But this doesn't mean you cannot work on it at your own rhythm.
chers,

**Comment by RyDroid** [comment]
Ok. Most of the stuff is easy and quick. However in this case, the copyright part will be a nightmare (it is not complicated but it will be long).

**Comment by Bertram25** [comment]
That's where @anholt may help since he made 90% of the job, and I hope he didn't lose his mind because if that. After all, he is for me, one of the stakeholders of sanity in the mesa driver development ;)

**Comment by RyDroid** [comment]
Ok, I will wait that he answers on GitHub, and sends him an email if he does not after a month or more.

**Comment by RyDroid** [comment]
When I compile with `debuild` on Debian GNU/Linux 9, I have this warning, but I have no idea how to fix it.

```
[  0%] Building CXX object src/CMakeFiles/valyriatear.dir/common/global/battle_media.cpp.o
cd /home/rydroid/workspace/ValyriaTear/obj-x86_64-linux-gnu/src && /usr/bin/c++    -I/home/rydroid/workspace/ValyriaTear/src -I/usr/include/SDL2 -I/usr/include/AL -I/usr/include/lua5.1 -I/home/rydroid/workspace/ValyriaTear/src/luabind  -g -O2 -fdebug-prefix-map=/home/rydroid/workspace/ValyriaTear=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -Wall -Wextra -std=c++11   -DPACKAGE_VERSION=\"0.1.0\" -DPKG_DATADIR=\"/usr/share/valyriatear/\" -DLOCALEDIR=\"/usr/share/locale/\" -DPKG_NAME=\"valyriatear/\" -DUSE_X11 -DLUABIND_USE_CXX11  -Winvalid-pch -o CMakeFiles/valyriatear.dir/common/global/battle_media.cpp.o -c /home/rydroid/workspace/ValyriaTear/src/common/global/battle_media.cpp
/home/rydroid/workspace/ValyriaTear/src/common/global/battle_media.cpp:17:29: warning: /home/rydroid/workspace/ValyriaTear/src/utils/utils_pch.h.gch: created with -gnone, but used with -gdwarf-2
 #include "utils/utils_pch.h"
                             ^
```

I suppose that the `.tx` is useless to rebuild and just a way to sync with transifex, am I right? If yes, is there a command line mechanism to sync without rights on transifex? Not for me: `echo 'tar-ignore = ".tx/"' >> debian/source/options` if yes to 1 and no to to 2.

Note for me: `echo 'tar-ignore = "src/luabind"' >> debian/source/options` + do a separate source package (valyria-tear-lua) + add it as a dependency for valyria-tear.

**Comment by bmorel** [comment]
Long time I have not compiled valyria, but the file is a precompiled header (gch: gnu compiled header I think). That kind of file is only really useful for people compiling often to speed up things. I doubt it have any use for a debia. source package, so I ´d say that disabling their generation would not hurt.
Also I would guess that the options are about debugging informations: -g is for debug compilation, and dwarf is a kind of binary format AFAIK.
In short, this warn is: gcc specific, about accelerating debugging build if I am not wrong. You may fix it through cmake parameters ( or use ccmake to have a nicer interface ofc).
I hope this will help you find directions.

**Comment by RyDroid** [comment]
It is strange: `make install` (with the makefile generated by CMake) is good, but dh does not use it (not for me: `dh $@ --builddirectory=build` might fix the issue), so checkinstall does a better package. o_O

@bmorel Ok, thanks. It is generated by the makefile of CMake, great.

**Comment by RyDroid** [comment]
I rebased and did some minor improvements on my branch. https://github.com/RyDroid/ValyriaTear/tree/debian-package-start
I am now trying to fix the 2 problems with `dh`: use install rule and remove the cause of the warning related to `src/utils/utils_pch.h`.
